test/jdk/java/awt/Robot/ScreenCaptureRobotTest.java

```diff
@@ -0,0 +1,173 @@
+ * and the source image are same.
+ * @run main/othervm -Dsun.java2d.uiScale=1 ScreenCaptureRobotTest
+ */
+public class ScreenCaptureRobotTest {
+
+    private static Frame frame;
+    private static volatile Canvas canvas;
+    private static BufferedImage realImage;
+    private static final int IMAGE_WIDTH = 200;
+    private static final int IMAGE_HEIGHT = 100;
+    private static volatile Point point;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            EventQueue.invokeAndWait(ScreenCaptureRobotTest::initializeGUI);
+            doTest();
+        } finally {
+            EventQueue.invokeAndWait(ScreenCaptureRobotTest::disposeFrame);
+        }
+    }
+
+    private static void initializeGUI() {
+        frame = new Frame("ScreenCaptureRobotTest Frame");
+        realImage = GraphicsEnvironment.getLocalGraphicsEnvironment()
+                .getDefaultScreenDevice().getDefaultConfiguration()
+                .createCompatibleImage(IMAGE_WIDTH, IMAGE_HEIGHT);
+
+        Graphics g = realImage.createGraphics();
+        g.setColor(Color.YELLOW);
+        g.fillRect(0, 0, IMAGE_WIDTH, IMAGE_HEIGHT);
+        g.setColor(Color.red);
+        g.setFont(new Font(Font.SANS_SERIF, Font.BOLD, 20));
+        g.drawString("Capture This", 10, 40);
+        g.dispose();
+
+        canvas = new ImageCanvas();
+        canvas.setBackground(Color.YELLOW);
+        frame.setLayout(new BorderLayout());
+        frame.add(canvas);
+        frame.setSize(300, 200);
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    private static void doTest() throws Exception {
+        Robot robot;
+        robot = new Robot();
+        robot.delay(500);
+        robot.waitForIdle();
+
+        EventQueue.invokeAndWait(() -> {
+            point = canvas.getLocationOnScreen();
+        });
+
+        Rectangle rect = new Rectangle(point.x + 10, point.y + 10, IMAGE_WIDTH,
+                IMAGE_HEIGHT);
+
+        BufferedImage capturedImage = robot.createScreenCapture(rect);
+
+        if (!compareImages(capturedImage, realImage)) {
+            String errorMessage = "FAIL : Captured Image is different from "
+                    + "the actual image";
+            System.err.println("Test failed");
+            saveImage(capturedImage, "CapturedImage.png");
+            saveImage(realImage, "RealImage.png");
+            throw new RuntimeException(errorMessage);
+        }
+    }
+
+    private static boolean compareImages(BufferedImage capturedImg,
+            BufferedImage realImg) {
+        int capturedPixel;
+        int realPixel;
+        int imgWidth;
+        int imgHeight;
+
+        imgWidth = capturedImg.getWidth(null);
+        imgHeight = capturedImg.getHeight(null);
+
+        if (imgWidth != realImg.getWidth()
+                || imgHeight != realImg.getHeight()) {
+            System.out.println(
+                    "Captured image and real image widths are different");
```

Comment:
Check is for width and height both, so let's have a generic message
```suggestion
                    "Captured image and real image are different in size");
```